### PR TITLE
Third branch

### PR DESCRIPTION
**Pull request by JemKrc**

### CODE REVIEW

File: force-app/main/default/classes/ExtensionControllerTest.cls [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    @isTest 
+    private static void testsearchSuccess()
+    {   
+        initData('testMethodNameSuccess');
+        
+        
+        List<Contact> Cons = [Select Id, LastName, Is_Primary_Contact__c FROM Contact WHERE LastName Like '%TestCon%'];
+        
+         
+        String sText = 'TestCon';
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/ExtensionControllerTest.cls:87]: Maybe you could rewrite your test method and have 1 Contact that does not start iwth "TestCon" ?
So, you check in a result if not ALL contacts are in search result (not all inserted ones, but only those that are OK for search conditions)

File: force-app/main/default/classes/ExtensionControllerTest.cls [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        
+        //Feeding with null Map<Integer, Contact>
+        Test.startTest();
+        
+        ext.search();
+        
+        Test.stopTest();
+        
+        //Assertion
+        System.assertEquals(4, ext.extContacts.size());
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/ExtensionControllerTest.cls]: I expect in fail scenario to see empty list or smth like that. Right now it looks the same as success scenario

File: force-app/main/default/classes/ExtensionControllerTest.cls [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        
+        Test.stopTest();
+        
+        //Assert
+        System.assertEquals(4,ext.extContacts.size());
+    }
+    
+    //Test that will fail
+     @isTest
+    private static void testMethodNameFailure() {
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/ExtensionControllerTest.cls]: please rename it to smth that makes sense

File: force-app/main/default/classes/extentionController.cls [hunk 4]
```diff
@@ -0,0 +1,71 @@
+public class ExtentionController {
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/extentionController.cls]: For future - please name controller with smth that makes more sense, for example
PrimaryContactController

File: force-app/main/default/classes/extentionController.cls [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+public List<contact> extContacts{get;set;}
+public String searchText{get
+    {if(searchText == null) searchText = ''; return searchText;}
+        set;}
+
+	//Constructing extension controller for standard account controller
+    public ExtentionController(ApexPages.StandardController stdCont){
+        this.acct = (Account)stdCont.getRecord();
+        try{
+        this.extContacts = [SELECT id, LastName, Title, Phone, Is_Primary_Contact__c FROM contact WHERE accountid = :acct.id ORDER BY LastName LIMIT 990];
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/extentionController.cls]: this query looks quite similar to the one in "search" method. Could you refactor your code to use 1 method for 2 queries or even to use 1 single query in 1 method just called twice?

File: force-app/main/default/classes/extentionController.cls [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            this.extContacts = new List<Contact>();
+        }
+        else{
+            this.extContacts.clear();
+        }
+        
+        //setting searchText property for query
+        String contLike = '%'+searchText+'%';
+        try{
+        this.extContacts = [Select LastName, Title, Phone, Is_Primary_Contact__c from Contact where accountid = :acct.id AND LastName like :contLike Order By LastName LIMIT 990];
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/extentionController.cls]: please write queries like
SELECT fields 
FROM object
WHERE conditions

File: force-app/main/default/classes/extentionController.cls [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    public void setPrimary(){
+        //Getting contactId from the page
+        String contactId = Apexpages.currentpage().getParameters().get('contactId');
+        
+        //Setting selected contact's field
+        Contact primaryContact = new Contact(Id = contactId, Is_Primary_Contact__c = true);
+        
+        //Getting the rest of the contacts that are not primary
+        List <Contact> conts = [SELECT Id From Contact WHERE Id != :primaryContact.Id];
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/extentionController.cls]: add a condition there
AND AccountId = : this.acct.Id

File: force-app/main/default/classes/extentionController.cls (hunk 6 above)
Comment (romanbychkoosf) [reply force-app/main/default/classes/extentionController.cls]: Please use this. for all class variables (not static)

File: force-app/main/default/classes/extentionController.cls [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        //To update previously assigned primary contact to non primary
+        for(Contact notPrimary: conts){
+            notPrimary.Is_Primary_Contact__c = false;
+        }
+        
+        //Combine all the contacts together
+        conts.add(primaryContact);
+        
+        try{
+        update conts;
```
Comment (romanbychkoosf) [inline force-app/main/default/classes/extentionController.cls]: please rewrite the code following Best Practices
try _SPACE_ {
_TAB_ code ;
}_SPACE_catch(Exception ex)_SPACE_{

**Review by romanbychkoosf — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/ExtensionControllerTest.cls (hunk 1 above)
Comment (JemKrc) [reply force-app/main/default/classes/ExtensionControllerTest.cls:87]: Added a contact to Cons List with different LastName. Worked. Passed the test

File: force-app/main/default/classes/ExtensionControllerTest.cls (hunk 2 above)
Comment (JemKrc) [reply force-app/main/default/classes/ExtensionControllerTest.cls]: Tried two asserts. One to pass for an empty List. One to fail with the same condition

File: force-app/main/default/classes/ExtensionControllerTest.cls (hunk 3 above)
Comment (JemKrc) [reply force-app/main/default/classes/ExtensionControllerTest.cls]: Done

File: force-app/main/default/classes/extentionController.cls (hunk 4 above)
Comment (JemKrc) [reply force-app/main/default/classes/extentionController.cls]: Changed to PrimaryContactController

File: force-app/main/default/classes/extentionController.cls (hunk 5 above)
Comment (JemKrc) [reply force-app/main/default/classes/extentionController.cls]: Created a new method. Used it in constructor, and in Search Method.

File: force-app/main/default/classes/extentionController.cls (hunk 6 above)
Comment (JemKrc) [reply force-app/main/default/classes/extentionController.cls]: Done

File: force-app/main/default/classes/extentionController.cls (hunk 7 above)
Comment (JemKrc) [reply force-app/main/default/classes/extentionController.cls]: Done

File: force-app/main/default/classes/extentionController.cls (hunk 8 above)
Comment (JemKrc) [reply force-app/main/default/classes/extentionController.cls]: Done

**Pull request closed without merging**